docs/topics/spiders.rst

```diff
@@ -261,6 +261,32 @@ to give data more structure you can use :class:`~scrapy.Item` objects::
 
 .. _spiderargs:
 
+    .. method:: update_settings(settings)
+
+        This is a classmethod which is called to override the configuration of your spider.
+        It receives the settings and passes :attr:`custom_settings` or `{}` to ``setdict`` 
+        method to register the setting with priority ``spider``. 
+        It is preferable to use the :attr:`custom_settings` when making spiders class 
+        hierarchies as parent since :attr:`custom_settings` is shadowed but parent
+        :meth:`update_settings` method can be called.
+
+        :param settings: scrapy settings for the configuration of internal components
+        :type args: dict
+Let's see an example::
+    import scrapy
+
+
+    class MySpider(scrapy.Spider):
+        name = 'example.com'
+        allowed_domains = ['example.com']
+        start_urls = ['http://www.example.com/feed.csv']
+        spider_settings = {'TEST1': 'spider', 'TEST2': 'spider'}
+        project_settings = {'TEST1': 'project', 'TEST3': 'project'}
+        self.spider_class.custom_settings = spider_settings
```

Comment:
This code won't work.

Reply:
Do I need to provide a code example ?

Reply:
You've already provided a code example, but it's incorrect.

Reply:
@wRAR Fixed the example. Please have a look into it.

Reply:
Sorry but this code makes no sense. It also doesn't even use `update_settings()`.

Reply:
I tried to show an example of how to update the settings ( I included update_settings()). if the example is not needed, I can remove the example altogether.

Reply:
This is not even an example of defining the `update_settings()` method.

Reply:
I  got this example from the unit tests. So, I guess the unit tests are wrong and needs to be configured too.

Reply:
Sorry but that's the wrong place to get an example (I would even say that if you don't know how to use this method you shouldn't write docs and examples for it).